src/components/field-zone/field-zone.repository.ts

```diff
@@ -69,25 +75,34 @@ export class FieldZoneRepository extends DtoRepository(FieldZone) {
     return await query.first();
   }
 
-  async readOne(id: ID, session: Session) {
+  async readOne(id: ID, session: Session): Promise<UnsecuredDto<FieldZone>> {
     const query = this.db
       .query()
       .apply(matchRequestingUser(session))
       .match([node('node', 'FieldZone', { id: id })])
-      .apply(matchPropList)
-      .optionalMatch([
-        node('node'),
-        relation('out', '', 'director', { active: true }),
-        node('director', 'User'),
-      ])
-      .return('propList, node, director.id as directorId')
-      .asResult<
-        StandardReadResult<DbPropsOfDto<FieldZone>> & {
-          directorId: ID;
-        }
-      >();
+      .apply(this.hydrate());
 
-    return await query.first();
+    const result = await query.first();
+    if (!result) {
+      throw new NotFoundException('Could not find field zone');
+    }
```

Comment:
I like converting the null to an exception here. For any place that you've added it here, can you remove it from the service? https://github.com/SeedCompany/cord-api-v3/blob/e1635597e3d5086ebd397b7a231625a6c8c925c2/src/components/field-zone/field-zone.service.ts#L88-L90